include/small/region.h

```diff
@@ -91,10 +105,33 @@ struct region
 static inline void
 region_create(struct region *region, struct slab_cache *cache)
 {
+	memset(region, 0, sizeof(*region));
 	region->cache = cache;
 	slab_list_create(&region->slabs);
 }
 
+/**
+ * Set callbacks for the region API.
+ *
+ * on_alloc will be called on any allocation.
+ * on_truncate will be called on any truncation.
+ * opaque is passed as callbacks argument.
+ *
+ * region_used is safe to use inside the callbacks. Inside on_alloc it
+ * will return usage before allocation. Inside on_truncate it will
+ * return usage after truncation.
+ */
+static inline void
+region_set_callbacks(struct region *region,
```

Comment:
Please add a unit test for the new functionality.

Reply:
Done.